### PR TITLE
Define kintone field type as a TypeScript type definition file.

### PR DESCRIPTION
**Pull request by yokotaso**
fix https://github.com/kintone/dts-gen/issues/116

Type geration tool generate type definition directory.
It is not good for Reuse type definition in dts-gen user production code.

But Subtable field is a little difficult. So, In this pull request I don't mind subtable field type.
By this modification, User can more easily read subtable type definition!

### CODE REVIEW

**Comment by koba04** [comment]
@yokotaso Thank you! This is amazing. I'll review this!! 👀

File: src/templates/expressions/fields.test.ts [hunk 1]
```diff
@@ -160,22 +118,10 @@ describe("SubTableField", () => {
...
-    value: {
-        contentType: string;
-        fileKey: string;
-        name: string;
-        size: string;
-    }[];
-    disabled?: boolean;
-    error?: string;
-};`.trim()
+"fieldName" : kintone.types.fields.File;`.trim()
```
Comment (zaki-yama) [inline src/templates/expressions/fields.test.ts]: It looks like you deleted unnecessary line break in 0ec46dc , so what about deleting this, too?

File: src/templates/expressions/typescriptfieldtypeconverter.ts [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+    DATE: "kintone.types.fields.Date",
+    NUMBER: "kintone.types.fields.Number",
+    DATETIME: "kintone.types.fields.DateTime",
+    TIME: "kintone.types.fields.Time",
+    DROP_DOWN: "kintone.types.fields.DropDown",
+    LINK: "kintone.types.fields.Link",
+    RADIO_BUTTON: "kintone.types.fields.RadioButton",
+};
+
+function convertStringField(typeName: string) {
```
Comment (zaki-yama) [inline src/templates/expressions/typescriptfieldtypeconverter.ts]: Do you have any reasons to group field types ( `KintoneStringFieldType...`, `KintoneStringListFieldType...`, etc.) ?
I thought they could be combined into one object, then we need to define just one `convert` function.

Comment (yokotaso) [reply src/templates/expressions/typescriptfieldtypeconverter.ts]: I agree with you. I fixed to simplify converter which convert from kintone field type to typescript defined field type.

File: src/templates/expressions/typescriptfieldtypeconverter.ts [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    MULTI_SELECT: "kintone.types.fields.MultiSelect",
+    RECORD_NUMBER: "kintone.types.fields.RecordNumber",
+    CREATED_TIME: "kintone.types.fields.CreatedTime",
+    UPDATED_TIME: "kintone.types.fields.UpdatedTime",
+    MODIFIER: "kintone.types.fields.Modifier",
+    CREATOR: "kintone.types.fields.Creator",
+    USER_SELECT: "kintone.types.fields.UserSelect",
+    GROUP_SELECT: "kintone.types.fields.GroupSelect",
+    ORGANIZATION_SELECT:
+        "kintone.types.fields.OrganizationSelect",
```
Comment (yokotaso) [inline src/templates/expressions/typescriptfieldtypeconverter.ts]: If I added File Field and Calc Field, code would be more simple. I'll try

File: src/templates/expressions/fields.ts [hunk 4]
```diff
@@ -2,14 +2,15 @@ import {
...
-        private stringFields: StringField[],
-        private calculatedFields: CalculatedField[],
-        private stringListFields: StringListField[],
-        private entityListFields: EntityListField[],
-        private fileFields: FileField[]
+        private stringFields: TSDefinedField[],
+        private calculatedFields: TSDefinedField[],
+        private stringListFields: TSDefinedField[],
+        private entityListFields: TSDefinedField[],
+        private fileFields: TSDefinedField[]
```
Comment (yokotaso) [inline src/templates/expressions/fields.ts]: I can simplify this class and fieldtype-convertert.ts, 
But If I did this modification, PR is too big...

So, I improve code with another modification.

File: kintone.d.ts [hunk 5]
```diff
@@ -255,4 +255,175 @@ declare namespace kintone {
 
     function getLoginUser(): LoginUser;
     function getUiVersion(): 1 | 2;
+
+    namespace types.fields {
```
Comment (koba04) [inline kintone.d.ts]: @yokotaso 
`dts-gen` generates a type definition for an app under the namespace of `kintone.types.Fields` even though we can change the namespace by `--namespace` option.
So this namespace(`kintnoe.types.fields`) might be confusing.

- `kintone.types.fields`
- `kintone.types.Fields`

Is there any other namespace for this?

File: src/templates/expressions/typescriptfieldtypeconverter.ts [hunk 6]
```diff
@@ -0,0 +1,42 @@
+const KintoneStringFieldTypeAndTypeScriptFieldTypeName = {
```
Comment (zaki-yama) [inline src/templates/expressions/typescriptfieldtypeconverter.ts]: Now that you merge all field types into one object, is the term "String" (used in `KintoneString...` ) still necessary?

File: src/templates/expressions/fields.ts [hunk 7]
```diff
@@ -23,124 +24,18 @@ ${toTsExpressions(this.fileFields)}
     }
 }
 
-export class StringField implements TsExpression {
+export class TSDefinedField implements TsExpression {
```
Comment (zaki-yama) [inline src/templates/expressions/fields.ts]: `TS` might be `Ts` (`s` is lowercase)? If you adjust the class name to other TypeScript-related variables like `TsExpression`.

**Comment by zaki-yama** [comment]
@yokotaso 
Sorry I added 2 more comments about the detailed matters.
And @koba04 also commented. Please see when you have time.

**Comment by yokotaso** [comment]
@koba04 @zaki-yama Sorry for late replay. I fix it. Thank you.

**Comment by zaki-yama** [comment]
@yokotaso 
I merged. Thank you!

**Pull request merged**